-lオプションの実装

konzo-goriki opened the pull request:
lsコマンドに-lオプションを実装した。

konzo-goriki committed:
-lオプションの実装

lsコマンドに-lオプションを実装した。

konzo-goriki commented on 05.ls/lib/ls.rb:
```diff
@@ -15,17 +61,165 @@ def show_files(files, len)
+  symbol = []
+
+  # user
+  symbol << if (mode & S_IRUSR).zero?
```
```
symbol = []
```

の宣言を省き、
```
symbol << if (mode & S_IRUSR).zero?
```
の行を
```
symbol = if (mode & S_IRUSR).zero?
```
とすればrubocopの警告通りメソッドの行数を20行以内に収めることが可能ですが、コードに統一性をもたせたいと考え修正を保留しました。

konzo-goriki commented on 05.ls/lib/ls.rb:
```diff
@@ -15,17 +61,165 @@ def show_files(files, len)
+  symbol = []
+
+  # group
+  symbol << if (mode & S_IRGRP).zero?
```
同上です。

konzo-goriki commented on 05.ls/lib/ls.rb:
```diff
@@ -15,17 +61,165 @@ def show_files(files, len)
+def convert_strmode_other(mode)
+  symbol = []
+  # other
+  symbol << if (mode & S_IROTH).zero?
```
同上です。

u1tnk commented on 05.ls/lib/ls.rb:
```diff
@@ -15,17 +61,165 @@ def show_files(files, len)
+  end
+end
+
+def convert_strmode_user(mode)
```
rubocopでひっかかってるのはここですか？
僕も業務でこれ以上短くする必要無いと判断することはありますが、プラクティスなので練習と思って警告消しましょう。

メソッドを短くするアイデアとしては

- if/elseのところを三項演算子とする
- 定数でhashを作ってcase/whenの替わりに利用する。以下が例です。変数名は適当

HOGE_HASH = {
  0 => "-",
  S_IXUSR => "x",
...
}
symbol << HOGE_HASH[mode & (S_IXUSR | S_ISUID)]
```

という感じにすると case/whenが不要になります。

u1tnk commented on 05.ls/lib/ls.rb:
```diff
@@ -15,17 +61,165 @@ def show_files(files, len)
+
+  puts "total #{count_blocks(files)}"
+
+  files.map do |file|
```
printするだけの処理であれば map では無く eachで良いですね。

u1tnk commented on 05.ls/lib/ls.rb:
```diff
@@ -15,17 +61,165 @@ def show_files(files, len)
+  if options['-l']
+    show_long_listing_format_files(files)
+  else
+    sliced_files = []
```
option lのときは show_long_listing_format_filesにまとまっているので、こちらもshow_files(名前も変えた方が良いですね)に表示するための処理をまとめたいですね。

konzo-goriki committed:
ファイルモード変換メソッド内のcase/whenをハッシュに置き換え

ファイルモードを数値から文字記号へ変換するためのメソッドを簡潔にするため
メソッド内のcase/whenをハッシュに置き換えた。

konzo-goriki committed:
ファイルモード変換メソッド内のif/elseを三項演算子で置き換え

ファイルモードを数値から文字記号へ変換するためのメソッドを簡潔にするため
メソッド内のif/else三項演算子に置き換えた。

konzo-goriki committed:
不必要なmapをeachに修正

filesに対する意味のないmapをeachに修正した。

konzo-goriki committed:
メインメソッド内のオプションなしでのlsコマンド実行処理を別メソッドにまとめた

メインメソッド内で処理していたオプション無しでのlsコマンド実行処理を
show_normal_format_filesメソッドにまとめた。

konzo-goriki committed:
convert_strmode_file_typeメソッド内のcase/whenをハッシュに置き換え

convert_strmode_file_typeメソッドを簡潔にするため
メソッド内のcase/whenをハッシュに置き換えた。

konzo-goriki reviewed (commented):
コメントがPending状態だと他人に見えないのを知りませんでした…すいません……。😥

u1tnk commented on 05.ls/lib/ls.rb:
```diff
@@ -4,28 +4,174 @@
+S_IFREG = '100000'.to_i(8)
+S_IFLNK = '120000'.to_i(8)
+S_IFSOCK = '140000'.to_i(8)
+CONV_MODE_FILE_TYPE_HASH = {
```
- 型名を変数/定数名にするのは極力避けたい
- hashも悪くは無いのですが、https://www.otsuka-shokai.co.jp/words/hash.html  的な意味もあるので、マッピング的な意味で _MAP、データテーブル的な意味で _TABLE などの方が比較的良いと思います。
    - メンター何人にどうしてるか聞いた結果です。
    - xxx_TYPES と複数形にするパターンも聞きました。私もやりますが、配列かな？と思われる事もあるので参考まで。


また CONV は略さない方が良いですね。 https://blog.jnito.com/entry/2020/10/20/092724 
というか、CONVは無くても意味が通りそうなので無しで大丈夫です。

konzo-goriki replied:
変数名についての指摘確認しました。
xxx_HASHはxxx_TABLEに修正しました。

また、CONVについては省略する方向で修正しました。
今後は省略形については気をつけて使っていきます。

u1tnk reviewed (changes requested):
変数名は大事なので、確認、修正お願いします 🙏

konzo-goriki committed:
定数名を修正(xxx_HASH -> xxx_TABLE, CONV_xxx -> xxx)

定数名を一部修正した。
xxx_HASHはデータテーブル的意味合いでxxx_TABLEに。
CONV_xxxはxxxでも意味が通るということでCONV_を省略。
